### ISSUE TITLE
Unintuitive error when building with esbuild

### ISSUE THREAD

**Issue by RaniSputnik** [Cazoo-uk/ssmble#17]
I found the following when trying to run [a simple project](https://github.com/Cazoo-uk/ssmble-esbuild-broken) built with esbuild. Here's the error:

```sh
(node:66036) UnhandledPromiseRejectionWarning: TypeError: reader.parser is not a function
    at setPropertiesFromData (/Users/ryan.loader/Code/experiments/ssmble-esbuild/index.js:867:36)
    at readRawData (/Users/ryan.loader/Code/experiments/ssmble-esbuild/index.js:888:20)
    at /Users/ryan.loader/Code/experiments/ssmble-esbuild/index.js:906:50
    at getConfig2 (/Users/ryan.loader/Code/experiments/ssmble-esbuild/index.js:483262:12)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async getCommerceToolsClientSecret (/Users/ryan.loader/Code/experiments/ssmble-esbuild/index.js:483317:20)
```

The reason this fails is because esbuild doesn't support the `emitDecoratorMetadata` compiler option, but I imagine you would also get this error if you hadn't enabled that option in the `tsconfig`.

Evan has said that esbuild [won't support any kind of type checking](https://esbuild.github.io/faq/#upcoming-roadmap) and given that he has ruled out producing typescript definition files, I imagine there are no plans to support the `emitDecoratorMetadata` option. 

I think it's worth making some documentation improvements:

* Throw a better error message in this situation _(we should confirm whether this also happens when the tsconfig is missing the `emitDecoratorMetadata` option)_.
* Update the documentation to mention that esbuild is explicitly not supported.

**Comment by RaniSputnik** [comment]
I just saw https://github.com/cazoo-engineering/ssmble/pull/16, maybe if that lands esbuild would be supported? @bobthemighty

**Comment by bobthemighty** [comment]
It's extraordinary that this issue was only opened on the 15th of March this year, it seems like decades ago. Regardless, I have tested your sample project with the latest version and seen it build successfully with my own eyes, therefore I am marking this one fixed by #16 .
